client/src/components/ArtistListElementCard.tsx

```diff
@@ -1,42 +1,103 @@
-import { FC } from 'react';
+import { FC, useState, useEffect } from 'react';
 import { Artist } from 'lib/artist';
 import { Card, CardContent, Typography, Grid } from '@mui/material';
 import FollowButton from 'components/FollowButton';
+import {
+  FollowArtistInput,
+  followArtistMutation,
+  User,
+  unfollowArtistMutation,
+} from 'lib/user';
+import { useMutation } from '@apollo/client';
+import toast from 'react-hot-toast';
 
 type ArtistListElementCardProps = {
   artist: Artist;
-  currentUserArtists: Set<Artist>;
+  currentUserArtistsName: Set<string>;
   loading: boolean;
   currentUserName: string;
 };
 
 const ArtistListElementCard: FC<ArtistListElementCardProps> = ({
   artist,
-  currentUserArtists,
+  currentUserArtistsName,
   loading,
   currentUserName,
-}) => (
-  <Grid item xs={6}>
-    <Card sx={{ width: 550, backgroundColor: '#e1f5fe', m: 2 }}>
-      <CardContent>
-        <Grid container>
-          <Grid item sm={8}>
-            <Typography variant="h5">{artist.name}</Typography>
-          </Grid>
-          <Grid item sm={4}>
-            {currentUserName && (
-              <FollowButton
-                artist={artist}
-                currentUserArtists={currentUserArtists}
-                loading={loading}
-                currentUserName={currentUserName}
-              />
-            )}
+}) => {
+  const [isFollow, setIsFollow] = useState<boolean>(false);
+  const [isFollowLoading, setIsFollowLoading] = useState<boolean>(true);
+  const [follow, { loading: followLoading }] = useMutation<
+    { follow: User },
+    FollowArtistInput
+  >(followArtistMutation);
+  const [unfollow, { loading: unfollowLoading }] = useMutation<
+    { unfollow: User },
+    FollowArtistInput
+  >(unfollowArtistMutation);
+
+  const handleFollow = (userName: string, artistName: string) => {
+    follow({
+      variables: { userName, artistName },
+    })
+      .then((_) => {
+        setIsFollow(true);
+      })
+      .catch((e: Error) => {
+        toast.error(`${e.message}`);
+      });
+  };
```

Comment:
`currentUserName` と `artist.name` は props 使えるから
```suggestion
  const handleFollow = () => {
    follow({
      variables: { userName: currentUserName, artistName: artist.name },
    })
      .then((_) => {
        setIsFollow(true);
      })
      .catch((e: Error) => {
        toast.error(`${e.message}`);
      });
  };
```
で良さそう